x-pack/plugins/security_solution/server/lib/detection_engine/signals/single_bulk_create.ts

```diff
@@ -51,7 +51,10 @@ export const filterDuplicateRules = (
     if (doc._source.signal == null) {
       return true;
     } else {
-      return !doc._source.signal.ancestors.some((ancestor) => ancestor.rule === ruleId);
+      return !(
+        doc._source.signal.ancestors.some((ancestor) => ancestor.rule === ruleId) ||
+        doc._source.signal.rule.id === ruleId
```

Comment:
Since `signal.ancestors` no longer contains the rule id of the current signal so we need to check `signal.rule.id` as well as the ancestors.